packages/core/src/index.ts

```diff
@@ -716,12 +718,42 @@ class Hopr extends EventEmitter {
     return addresses.map((addr) => addr.multiaddr)
   }
 
+  /**
+   * Validates the manual intermediate path by checking if it does not contain
+   * channels that are not opened.
+   * Throws an error if some channel is not opened.
+   * @param intermediatePath
+   */
+  private async validateIntermediatePath(intermediatePath: PublicKey[]) {
+    // checking if path makes sense
+    for (let i = 0; i < intermediatePath.length; i++) {
+      let ticketIssuer: PublicKey
+      let ticketReceiver: PublicKey
+
+      if (i == 0) {
+        ticketIssuer = PublicKey.fromPeerId(this.getId())
+        ticketReceiver = intermediatePath[0]
+      } else {
+        ticketIssuer = intermediatePath[i - 1]
+        ticketReceiver = intermediatePath[i]
+      }
+
+      if (ticketIssuer.eq(ticketReceiver)) log(`WARNING: duplicated adjacent path entries.`)
```

Comment:
shouldn't we disallow adjacent path entries?

Reply:
I think this is resolved by your #4027 PR ? If yes, then I can simply remove this warning here.

Reply:
Yes, we fail the command in `hopr-admin` if there are adjacent nodes (1). I still think we should handle this on the protocol level too for API users.

Reply:
Ok, let's resolve this in a separate PR.